esmvaltool/diag_scripts/iht_toa/single_model_diagnostics.py

```diff
@@ -0,0 +1,717 @@
+    record = {
+        'statistics': ['mean'],
+        'domains': ['global'],
+        'plot_types': [plot_type],
+        'authors': [
+            'andela_bouwe',
+            'righi_mattia',
+        ],
+        'references': [
+            'acknow_project',
+        ],
+        'ancestors': ancestor_files,
+    }
+    return record
+
+
+def area_average(cube, latitude='latitude', longitude='longitude', mdtol=1):
+    """Return area-weighted average of a cube."""
+    if cube.coord(latitude).bounds is None:
+        cube.coord(latitude).guess_bounds()
+    if cube.coord(longitude).bounds is None:
+        cube.coord(longitude).guess_bounds()
+    grid_areas = iris.analysis.cartography.area_weights(cube)
+    cube_avg = cube.collapsed([longitude, latitude],
+                              iris.analysis.MEAN,
+                              weights=grid_areas,
+                              mdtol=mdtol)
+    return cube_avg
+
+
+def var_name_constraint(var_name):
+    """Shortcut to create constraint for variable name."""
+    return iris.Constraint(cube_func=lambda c: c.var_name == var_name)
+
+
+def call_poisson(flux_cube, latitude='latitude', longitude='longitude'):
+    """Top-level function that calls the Poisson solver for source cube."""
+    if flux_cube.coord(latitude).bounds is None:
+        flux_cube.coord(latitude).guess_bounds()
+    if flux_cube.coord(longitude).bounds is None:
+        flux_cube.coord(longitude).guess_bounds()
+
+    # Remove average of flux field to account for storage term
+    data = flux_cube.data.copy()
+    grid_areas = iris.analysis.cartography.area_weights(flux_cube)
+    data_mean = flux_cube.collapsed(["longitude", "latitude"],
+                                    iris.analysis.MEAN,
+                                    weights=grid_areas).data
+    data -= data_mean
+
+    logger.info("Calling spherical_poisson")
+    sphpo = SphericalPoisson(logger,
+                             source=data * (6371e3**2.0),
+                             tolerance=2.0e-4)
+    sphpo.solve()
+    sphpo.calc_mht()
+    logger.info("Ending spherical_poisson")
+
+    # Energy flux potential (P)
+    p_cube = flux_cube.copy()
+    p_cube.var_name = f"{flux_cube.var_name}_efp"
+    p_cube.long_name = f"energy_flux_potential_of_{flux_cube.var_name}"
+    p_cube.standard_name = None
+    p_cube.units = 'J s-1'
+    p_cube.data = sphpo.efp[1:-1, 1:-1]
+
+    # MHT data cube
+    mht_cube = flux_cube.copy()
+    mht_cube = mht_cube.collapsed('longitude', iris.analysis.MEAN)
+    mht_cube.var_name = f"{flux_cube.var_name}_mht"
+    mht_cube.long_name = f"meridional_heat_transport_of_{flux_cube.var_name}"
+    mht_cube.standard_name = None
+    mht_cube.units = 'W'
+    mht_cube.data = sphpo.mht
+
+    return p_cube, mht_cube
+
+
+def symmetry_metric(data, grid):
+    """Calculate symmetry metrics.
+
+    A perfectly symmetrical latitude band gives S=0. As coded, the
```

Comment:
It is also implicitly assuming a uniform grid spacing in latitude. It feels like it would be safer and more explicit to do cube.extract() for each lat band, which would also remove the multiple of 6 restriction. Is it worth exploring this option?

Reply:
Yes, the current approach is not very elegant, I'll try and code it in a more flexible way.